libsolidity/codegen/ir/IRGenerationContext.h

```diff
@@ -102,7 +104,13 @@ class IRGenerationContext
 
 	std::string newYulVariable();
 
-	std::string generateInternalDispatchFunction(YulArity const& _arity);
+	void initializeInternalDispatch(InternalDispatchMap _internalDispatchMap);
+	InternalDispatchMap consumeInternalDispatchMap();
+	bool internalDispatchClean() const { return m_internalDispatchMap.empty() && m_directInternalFunctionCalls.empty(); }
+
+	void internalFunctionCalledThroughDispatch(YulArity const& _arity);
```

Comment:
Can you add some documentation to these three function? 1-2 lines should be enough.

Reply:
Sure.

Reply:
Done.

Reply:
A bit longer (3-4 lines) but to shorten it I would have to cut the first sentence that says what the function does. It's is a bit redundant but still, the docstring feels a bit weird without it.